.pre-commit-config.yaml

```
@@ -0,0 +1,43 @@
+# Configuration file for pre-commit (https://pre-commit.com/).
+# Please run `pre-commit run --all-files` when adding or changing entries.
+
+repos:
+  - repo: local
+    hooks:
+      - id: black
+        name: black
+        entry: black
+        language: system
+        stages: [commit]
+        types: [python]
```

I'm sort of new to these configs, but I think we want this to run just the check (using the `--check` flag) so that it doesn't actually reformat the files as part of the CI.

```suggestion
        types: [python]
        args: [--check]
```

---

I'd prefer to leave this as is, for two reasons:

1. it means people can run a single command locally to do all the formatting, and
2. it's the last command in the CI job, so only the other pre-commit jobs depend on the output.

---

> 2\. it achieves the same thing in terms of linting (`pre-commit run` returns a non-zero exit code when anything is changed by any target),

Thanks, that's the piece I was missing. In that case this looks good to me.